provisioning/baremetal_pod.go

```diff
@@ -798,13 +782,29 @@ func EnsureMetal3Deployment(info *ProvisioningInfo) (updated bool, err error) {
 		info.EventRecorder, metal3Deployment, expectedGeneration)
 	if err != nil {
 		err = fmt.Errorf("unable to apply Metal3 deployment: %w", err)
-		return
+		// Check if ApplyDeployment failed because the existing Pod had an outdated
+		// Pod Selector.
+		selector, get_err := getMetal3DeploymentSelector(info.Client.AppsV1(), info.Namespace)
+		if get_err != nil || selector == metal3Deployment.Spec.Selector {
```

Comment:
Pointer comparison is unlikely to ever match here ;) Does LabelSelector have a DeepEqual() method?

Reply:
Yeah, duh! Yes, it does. Fixed!